core/src/main/java/org/apache/accumulo/core/util/threads/Threads.java

```diff
@@ -55,22 +55,26 @@ public static Runnable createNamedRunnable(String name, Runnable r) {
     return new NamedRunnable(name, r);
   }
 
-  public static Thread createThread(String name, Runnable r) {
-    return createThread(name, OptionalInt.empty(), r, UEH);
+  public static Thread createNonCriticalThread(String name, Runnable r) {
```

Comment:
I'm not sure we need to rename these to be NonCritical, I think the fact that we have a createCriticalThread implies that the others are non-critical.

Reply:
My goal was to force callers to always make a conscious decision about which method to use.

Reply:
> My goal was to force callers to always make a conscious decision about which method to use.

One possible way to achieve this is via an enum that must always be passed to the method instead of encoding the information int the method name. Feel free to ignore this comment, not sure it would be an improvement was just a thought based on the goal mentioned.


```java
   enum ThreadType {
      CRITICAL,
      NON_CRITICAL
   }

   public static Thread createThread(String name, ThreadType type,, Runnable r) {...}

``